chain/beacon/chainstore.go

```diff
@@ -39,22 +40,23 @@ type chainStore struct {
 }
 
 func newChainStore(l log.Logger, cf *Config, cl net.ProtocolClient, v *vault.Vault, store chain.Store, t *ticker) (*chainStore, error) {
-	// we make sure the chain is increasing monotonically
-	as, err := newAppendStore(store)
+	// we write some stats about the timing when new beacon is saved
+	ds := newDiscrepancyStore(store, l, v.GetGroup(), cf.Clock)
```

Comment:
Ohhh, we're changing the node's id in the logger during resharing:
https://github.com/drand/drand/blob/47fc7abf063eee475036fb07f0666e24ff4d7706/core/drand_beacon_control.go#L204
 (we're actually changing it a bit too early, since we change it prior to transition... but that's not the main issue), but we are not changing this logger, which is also passed to the SyncManager nor the one in the discrepancy logger passed here, which means we probably get mangled logs until the next time the node restarts after a ceremony :O

Reply:
For these two issues, I'd say it's probably better to open a separate issue and have a bit of design thinking on how the node id should be propagated when it's changed.
Currently, the logger has a bit of a race condition, which we avoid with properly placed locks, for now. But it doesn't necessarily feel like the right solution.

One approach I see as a solution for this would be to stop the ChainStore and SyncManager and restart them when the values change. That way there is no risk of fixing this problem but leaving other problems lingering, masked by what would seemingly work as expected.

This solution has the advantage of not using locks or adding even more to an already lock-heavy area.

What do you think?

Reply:
Sounds good, let's open an issue and merge this in the meantime

Reply:
Done: https://github.com/drand/drand/issues/1174